app/services/queue_listener.rb

```diff
@@ -66,9 +66,10 @@ def cancel_subscription(params)
     end
   end
 
-  def create_payment(params)
+  def upsert_payment(params)
     res = client.create_recurring_payment(params[:params])
-    unless res.success?
+    SubscriptionUpdater.run(params);
+    unless res.success? # AskOmar how to check success of the updater here?
```

Comment:
@osahyoun I have a ruby question if I want to check whether or not the `SubscriptionUpdater` ran successfully, can I do something like this:
```
updaterRes = SubscriptionUpdater.run(params);
unless res.success? && updaterRes?
```

Given that the SubscriptionUpdater look like this:

```
def run
      handle_subscription_charge()
    end
  
    private
  
    def handle_subscription_charge()
        return unless subscription
        # Return if a transaction record already exists for this subscription charge. This is to prevent processing
        # the same subscription charge twice.
        return unless Payment::Braintree::Transaction.find_by(transaction_id: @transaction_id).blank?

        update_subscription()
        create_subscription_charge()
        true
      end
```